### PR TITLE
Part 4: Update Graphs and Stats pages

### PR DESCRIPTION
**Pull request by QSparks**
This PR focuses on updates to graphs and statistical pages within the application. These changes are part of the larger fnlf-cmip6-release.

Key changes include:
- New baseline mean column on the stats page. Cell values include variable-specific formatting for units and precision.
- Graph tab hover popups display projected values, with units and precision.
- Graph tab baseline point displays the baseline median value
- Includes a new function to fetch median statistics from precalculated regional stat files.

Demo link: https://services.pacificclimate.org/dev/plan2adapt/app/

### CODE REVIEW

File: src/utils/variables-and-units.js [hunk 1]
```diff
@@ -99,24 +99,32 @@ export const unitsSuffix = units => {
...
-
-export const expToFixed = s => {
-  // Convert a string representing a number in exponential notation to a string
-  // in (nominally) fixed point notation. Why? Because `Number.toPrecision()`
-  // returns exponential notation frequently when we do not want it to. So
-  // we apply this.
-  const match = s.match(/-?\d\.\d+e[+-]\d+/);
-  if (!match) {
-    return s;
+// No longer used. setting precision with decimal places rather than sigfigs.
```
Comment (corviday) [inline src/utils/variables-and-units.js:102]: Why was this changed, I think sigfigs are more widely used? Just curious.

File: src/components/graphs/ChangeOverTimeGraph/ChangeOverTimeGraph.js [hunk 2]
```diff
@@ -32,6 +32,28 @@ class ChangeOverTimeGraphDisplay extends React.Component {
...
   // statistics that are fetched asynchronously, according to the
   // selected region and climatological time period.
+  constructor(props) {
+    super(props);
+    this.state = {
+      median: null
+    };
+  }
+
+  componentDidMount() {
```
Comment (corviday) [inline src/components/graphs/ChangeOverTimeGraph/ChangeOverTimeGraph.js:42]: All the other data this component uses seems to be fetched with `withAsyncData`. It's a little confusing to fetch just this one piece here. Is there a particular reason to do ti this way?

In any case, if you decide to keep the data fetch handled this way, edit the comments above to note that this is no longer a state-free component and perhaps note which data comes to the component via which pathways.

File: src/data-services/summary-stats.js [hunk 3]
```diff
@@ -22,9 +22,72 @@ export const fetchSummaryStatistics = (
...
+    .then(response => response.data);
+};
+
+
+const seasonToIndex = {
+  'winter': 0,
+  'spring': 1,
+  'summer': 2,
+  'fall': 3,
+  'annual': 0
```
Comment (corviday) [inline src/data-services/summary-stats.js:40]: maybe add a comment about why annual is in this list

File: src/data-services/summary-stats.js [hunk 4]
```diff
@@ -22,9 +22,72 @@ export const fetchSummaryStatistics = (
...
+      return null;
+    });
 };
+
+function parseCSV(data) {
+  const lines = data.split('\n');
+  const result = [];
+  const headers = lines[0].split(',').map(header => header.trim());
+
+  //console.log(`CSV headers: ${headers.join(', ')}`);
```
Comment (corviday) [inline src/data-services/summary-stats.js:78]: remove debug output?

**Review by corviday — APPROVED** [review]
Looks good! One minor question about react lifecycle and data fetching. Poked at the demo, everythign works! I didn't even notice the units were missing on the live graphs, great to have that fixed.

File: src/components/graphs/ChangeOverTimeGraph/ChangeOverTimeGraph.js (hunk 2 above)
Comment (QSparks) [reply src/components/graphs/ChangeOverTimeGraph/ChangeOverTimeGraph.js]: I agree that we should maintain consistent data fetching using withAsyncData. I have updated the relevant code to ensure the medians are fetched similarly to the summary statistics.

**Review by corviday — APPROVED** [review]
LGTM

**Comment by corviday** [comment]
Do you have a demo for this latest change I can play with? Code looks good, but it seems like a big change that could use a little testing.

**Comment by QSparks** [comment]
> Do you have a demo for this latest change I can play with? Code looks good, but it seems like a big change that could use a little testing.

Latest demo: https://services.pacificclimate.org/dev/plan2adapt/app/
There will also be updated text reflecting these changes in [PR Part 5](https://github.com/pacificclimate/plan2adapt-v2/pull/268).
Thanks!

**Comment by corviday** [comment]
The text on the "graphs" tab in the demo says:

> The black line indicates the mid-point (median) in the set.

Is this still true?

**Review by corviday — APPROVED** [review]
Code and demo look good. I'm a little confused about the text, but it sounds like there's already a PR addressing that?
